pkg/operators/cli/clioperator.go

```diff
@@ -259,10 +259,14 @@ func (o *cliOperatorInstance) PreStart(gadgetCtx operators.GadgetContext) error
 				}
 			}
 			headerFuncs = append(headerFuncs, printHeader)
-
+			
+			printedHeader := false
 			for _, headerFunc := range headerFuncs {
-				headerFunc()
-			}
+				if !printedHeader {
+					headerFunc()
+					printedHeader = true
+				}
+			}			
```

Comment:
I see two issues with this approach:

1. For `snapshot_socket` and `snapshot_process` gadgets, which don't require the terminal to be cleaned up (datasources don't have `AnnotationClearScreenBefore` annotation), the header is still being printed twice.
2. For `top_*` gadgets, if I set `--map-fetch-interval 10s`, the terminal will be cleaned up, but nothing will be printed for 10 seconds (of course, not even the header). I'm wondering if this may make users think IG is blocked or so.